lib/jsdoc/util/logger.js

```diff
@@ -130,34 +141,26 @@ function addPrefix(args, prefix) {
     return updatedArgs || args;
 }
 
-// TODO: document events
-function wrapLogFunction(name, func) {
-    var eventName = 'logger:' + name;
-    var upperCaseName = name.toUpperCase();
-    var level = LEVELS[upperCaseName];
-    var prefix = PREFIXES[upperCaseName];
-
-    return function() {
-        var loggerArgs;
 
-        var args = Array.prototype.slice.call(arguments, 0);
-
-        if (logLevel >= level) {
-            loggerArgs = addPrefix(args, prefix);
-            func.apply(null, loggerArgs);
-        }
+var Logger = function () {
+    this.handlers = {};
```

Comment:
Yes and no. I get how you want to use the logger which helps a lot. Nonetheless I think that users of the programmatic interface may want to get debugging data by just setting an option. They shouldn't have to require the logger and call setDefaultHandlers on their own to just get it working. 
Also your initial comment was about getting away with `this.handlers`. If I remove it along with setHandlers, how do you want to tie the console.* methods to the different log levels? Do you want to go back to something like the wrapLogFunction instead of the invoke private method? Do you want to use events like in my comment above?